src/dotnet/ShellShim/LinuxEnvironmentPath.cs

```diff
@@ -44,28 +44,27 @@ public void AddPackageExecutablePathToUserPath()
             }
 
             var script = $"export PATH=\"$PATH:{_packageExecutablePath.PathWithDollar}\"";
-            _fileSystem.WriteAllText(_profiledDotnetCliToolsPath, script);
+            _fileSystem.WriteAllText(DotnetCliToolsProfilePath, script);
         }
 
         private bool PackageExecutablePathExists()
         {
-            var environmentVariable = _environmentProvider
-                .GetEnvironmentVariable(PathName);
-
-            if (environmentVariable == null)
+            var value = _environmentProvider.GetEnvironmentVariable(PathName);
+            if (value == null)
             {
                 return false;
             }
 
-            return environmentVariable
-                .Split(':').Contains(_packageExecutablePath.Path);
+            return value
+                .Split(':')
+                .Any(p => p == _packageExecutablePath.Path || p == _packageExecutablePath.PathWithTilde);
```

Comment:
will ~/.dotnet work no Linux?

Reply:
So bash does support tilde expansion in `PATH`:

```
$ export PATH=$PATH:\~/foo
$ mkdir ~/foo
$ cat << \EOF >> ~/foo/bar
#! /usr/bin/env bash
echo hello world!
EOF
$ chmod +x ~/foo/bar
$ bar
hello world!
```

The intention with this change was to make sure we respected that.  However, this is very uncommon and certainly breaks tools like `which` that don't expand tilde.  `zsh` doesn't appear to expand tilde in `PATH` either.

I'm fine with requiring actual directory paths without tilde (i.e. backing out this change) if we think this isn't valuable to support.

Reply:
I don't have strong opinion, it is good to have. Just all things about PATH are hard to test and take a long time to know what's wrong. It looks fine. But in general it is better to reduce change around that area.